src/App.ts

```diff
@@ -160,7 +160,7 @@ export default class App {
     this.listeners = [];
 
     // Check for required arguments of ExpressReceiver
-    if (signingSecret !== undefined) {
+    if (signingSecret !== undefined && receiver === undefined) {
```

Comment:
We can change these lines to make it more easy-to-understand:
```js
// Check for required arguments of ExpressReceiver
if (signingSecret !== undefined && receiver === undefined) {
    this.receiver = new ExpressReceiver({ signingSecret, logger, endpoints });
} else if (receiver === undefined) {
    // Check for custom receiver
```
How about this?
```js
// Check for required arguments of ExpressReceiver
if (receiver !== undefined) {
    // NOTE: signingSecret would be ignored in this case
    this.receiver = receiver;
} else {
    // no custom receiver given here
    if (signingSecret === undefined) {
        throw errorWithCode(
            'Signing secret not found, so could not initialize the default receiver. ' +
            'Set a signing secret or use a custom receiver.',
            ErrorCode.AppInitializationError,
        );
    } else {
        this.receiver = new ExpressReceiver({ signingSecret, logger, endpoints });
    }
}
```